service/src/main/scala/com/yahoo/maha/service/calcite/MahaCalciteSqlParser.scala

```diff
@@ -132,12 +134,22 @@ case class DefaultMahaCalciteSqlParser(mahaServiceConfig: MahaServiceConfig) ext
     }
   }
 
-  def getCube(sqlNode: SqlNode, registry:Registry) : Option[PublicFact] = {
+  def getCube(fromTable: String, registry:Registry) : Option[PublicFact] = {
+    registry.getFact(fromTable)
+  }
+
+  def getFromTableAndTableSchema(sqlNode: SqlNode): (String, Option[String]) = {
     sqlNode match {
       case sqlIdentifier: SqlIdentifier =>
-        registry.getFact(sqlIdentifier.getSimple.toLowerCase)
-      case e=>
-        throw new IllegalArgumentException(s"missing case ${e} in get cube method")
+        val fromTableAndTableSchema = sqlIdentifier.toString.toLowerCase.split("\\.")
```

Comment:
Instead of string parsing, SqlIndentifier provides the clean way to get the schema, i see that it is populated as part of names. You can just check on the names array length to see if schema is present

[
<img width="558" alt="Screen Shot 2021-07-01 at 1 27 40 PM" src="https://user-images.githubusercontent.com/4935454/124185704-252f9600-da70-11eb-8440-f2acd83ca6fc.png">
](url)